src/main/rx/rx_unified_dsm.c

```diff
@@ -37,26 +36,52 @@ extern uint8_t ready_for_next_telemetry;
 #define USART usart_port_defs[serial_rx_port]
 #define SPECTRUM_BIND_PIN usart_port_defs[profile.serial.rx].rx_pin
 
-void rx_serial_process_dsmx() {
+static uint8_t dsm_protocol = 0;
+#define DSMX_11_2048 0xb2
+#define DSMX_22_2048 0xa2
+#define DSM2_11_2048 0x12
+#define DSM2_22_1024 0x01
+
+void rx_serial_process_dsm() {
 
   for (uint8_t counter = 0; counter < 16; counter++) {    //First up, get the rx_data out of the RX buffer and into somewhere safe
     rx_data[counter] = rx_buffer[counter % RX_BUFF_SIZE]; // This can probably go away, as long as the buffer is large enough
   }
 
-#ifdef RX_DSM2_1024_TEMP
-  float dsm2_scalefactor = (0.29354210f / DSM_SCALE_PERCENT);
-  // 10 bit frames
-  static uint8_t spek_chan_shift = 2;
-  static uint8_t spek_chan_mask = 0x03;
-  static uint8_t dsm_channel_count = 7;
-#else //DSMX_2048
-#define RX_DSMX_2048_UNIFIED
-  float dsmx_scalefactor = (0.14662756f / DSM_SCALE_PERCENT);
-  // 11 bit frames
-  static uint8_t spek_chan_shift = 3;
-  static uint8_t spek_chan_mask = 0x07;
-  static uint8_t dsm_channel_count = 12;
-#endif
+  if (dsm_protocol == 0) {        //dsm variant has not been selected yet
+    dsm_protocol = rx_buffer[1];  //detect dsm variant on first contact and run with it
+  }
+
+  uint8_t spek_chan_shift;
+  uint8_t spek_chan_mask;
+  uint8_t dsm_channel_count;
+  float dsm_scalefactor;
+  float dsm_offset;
+
+  if (dsm_protocol == DSMX_11_2048 || dsm_protocol == DSMX_22_2048){
+    // 11 bit frames
+    spek_chan_shift = 3;
+    spek_chan_mask = 0x07;
+    dsm_channel_count = 12;
+    dsm_scalefactor = (0.14662756f / DSM_SCALE_PERCENT);
+    dsm_offset = 1024.0f;
+  }
+  if (dsm_protocol == DSM2_11_2048){
+    // 10 bit frames
+    spek_chan_shift = 2;
+    spek_chan_mask = 0x03;
+    dsm_channel_count = 7;
+    dsm_scalefactor = (0.14662756f / DSM_SCALE_PERCENT);
+    dsm_offset = 1024.0f;
```

Comment:
the [spektrum datasheet](https://www.spektrumrc.com/ProdInfo/Files/Remote%20Receiver%20Interfacing%20Rev%20A.pdf) (Section 8) says DSM2 2048 also uses 11bit channels.
did you verify this?

Reply:
No, this configuration is unverified.   I am unable to test this particular configuration with any gear I have available.  The above combination was my best guess.

Reply:
all references i have found point towards it being 11bit as well, so lets go with that.